In [9]:
import os
import requests

week = ["181111", "181118", "181125", "181202", "181209", "181216", "181223", "181230", "190106", "190113"]
key = os.getenv("KOBIS_KEY")
url = f"http://www.kobis.or.kr/kobisopenapi/webservice/rest/boxoffice/searchWeeklyBoxOfficeList.json?key={key}&weekGb=0&targetDt=20"



In [10]:
kobis_data = []
for wk in week:
    kobis_data.append(requests.get(url + wk).json()["boxOfficeResult"])

In [11]:
kobis_data

[{'boxofficeType': '주간 박스오피스',
  'showRange': '20181105~20181111',
  'yearWeekTime': '201845',
  'weeklyBoxOfficeList': [{'rnum': '1',
    'rank': '1',
    'rankInten': '0',
    'rankOldAndNew': 'OLD',
    'movieCd': '20177538',
    'movieNm': '완벽한 타인',
    'openDt': '2018-10-31',
    'salesAmt': '14644885538',
    'salesShare': '45.8',
    'salesInten': '1121885627',
    'salesChange': '8.3',
    'salesAcc': '28553908349',
    'audiCnt': '1711146',
    'audiInten': '88553',
    'audiChange': '5.5',
    'audiAcc': '3375748',
    'scrnCnt': '1325',
    'showCnt': '44759'},
   {'rnum': '2',
    'rank': '2',
    'rankInten': '0',
    'rankOldAndNew': 'OLD',
    'movieCd': '20185485',
    'movieNm': '보헤미안 랩소디',
    'openDt': '2018-10-31',
    'salesAmt': '10252716988',
    'salesShare': '32.1',
    'salesInten': '3940138375',
    'salesChange': '62.4',
    'salesAcc': '16587971601',
    'audiCnt': '1133842',
    'audiInten': '427066',
    'audiChange': '60.4',
    'audiAcc': '1843165',
   

In [83]:
kobis_list = []
for kobis_d in kobis_data:
    for index, kobis in enumerate(kobis_d["weeklyBoxOfficeList"]):
        add = [kobis["movieCd"], kobis["movieNm"], kobis["audiAcc"], kobis_d["showRange"][9:-1]]
        if kobis_list == []:
            kobis_list.append(add)
        else:
            for idx in range(len(kobis_list)):
                if add[0] in kobis_list[idx]:
                    kobis_list[idx] = add
                    break
            else:
                kobis_list.append(add)
                
print(kobis_list)
print(len(kobis_list))

[['20177538', '완벽한 타인', '5270621', '2018121'], ['20185485', '보헤미안 랩소디', '9785643', '2019011'], ['20170513', '동네사람들', '445625', '2018111'], ['20181171', '바울', '229837', '2018120'], ['20174981', '창궐', '1588443', '2018111'], ['20010291', '해리포터와 마법사의 돌', '259733', '2018111'], ['20179006', '여곡성', '55997', '2018111'], ['20181404', '벽 속에 숨은 마법시계', '211233', '2018111'], ['20180523', '스타 이즈 본', '458917', '2018111'], ['20182693', '구스범스: 몬스터의 역습', '21400', '2018111'], ['20176814', '신비한 동물들과 그린델왈드의 범죄', '2403686', '2018120'], ['20183050', '번 더 스테이지: 더 무비', '298402', '2018112'], ['20184481', '성난황소', '1567264', '2018120'], ['20182935', '출국', '71407', '2018112'], ['20186822', '너의 췌장을 먹고 싶어', '48910', '2018112'], ['20189869', '해피 투게더', '15745', '2018111'], ['20183073', '베일리 어게인', '78439', '2018120'], ['20182669', '툴리', '24325', '2018112'], ['20177552', '국가부도의 날', '3723915', '2018122'], ['20181905', '후드', '276658', '2018120'], ['20183007', '거미줄에 걸린 소녀', '25172', '2018120'], ['20182966', '투 프렌즈', '20419

In [87]:
import csv

f_w = open("boxoffice.csv", "a+", encoding="utf-8", newline="")
writer = csv.writer(f_w)

for i in kobis_list:
    writer.writerow(i)

f_w.close()

- 영화 대표 코드 : movieCd

- 영화명 : movieNm

- 해당일 누적관객수 : audiAcc
    - 중복되면 최신 정보를 반영해야 한다.

- 해당일 : showRange
    - showRange는 "20181105~20181111"식이라서 맨 마지막 8자리를 구해야 할 듯.


movie_code, title, audience, recorded_at

영화 코드, 제목, 관객수, 기준일

위의 영어는 안 쓰고, 43줄이 있어야 함

# CLEAR!

In [ ]:
f_r = open("boxoffice.csv", "r", encoding="utf-8", newline="")
reader = csv.reader(f_r)

for line in reader:
    print(line)

f_r.close()

In [91]:
key = os.getenv("KOBIS_KEY")

f_r = open("boxoffice.csv", "r", encoding="utf-8", newline="")
reader = csv.reader(f_r)

movie_info = []
for line in reader:
    url = f"http://www.kobis.or.kr/kobisopenapi/webservice/rest/movie/searchMovieInfo.json?key={key}&movieCd={line[0]}"
    movie_info.append(requests.get(url).json())
    
    
f_r.close()

print(movie_info)



[{'movieInfoResult': {'movieInfo': {'movieCd': '20177538', 'movieNm': '완벽한 타인', 'movieNmEn': 'Intimate Strangers', 'movieNmOg': '', 'showTm': '115', 'prdtYear': '2018', 'openDt': '20181031', 'prdtStatNm': '개봉', 'typeNm': '장편', 'nations': [{'nationNm': '한국'}], 'genres': [{'genreNm': '드라마'}, {'genreNm': '코미디'}], 'directors': [{'peopleNm': '이재규', 'peopleNmEn': 'LEE Jae-kyoo'}], 'actors': [{'peopleNm': '유해진', 'peopleNmEn': 'YOO Hae-jin', 'cast': '', 'castEn': ''}, {'peopleNm': '조진웅', 'peopleNmEn': 'CHO Jin-woong', 'cast': '', 'castEn': ''}, {'peopleNm': '이서진', 'peopleNmEn': 'LEE Seo-jin', 'cast': '준모', 'castEn': ''}, {'peopleNm': '염정아', 'peopleNmEn': 'YUM Jung-ah', 'cast': '', 'castEn': ''}, {'peopleNm': '김지수', 'peopleNmEn': 'KIM Ji-soo', 'cast': '', 'castEn': ''}, {'peopleNm': '송하윤', 'peopleNmEn': 'SONG Ha-yoon', 'cast': '', 'castEn': ''}, {'peopleNm': '윤경호', 'peopleNmEn': '', 'cast': '', 'castEn': ''}, {'peopleNm': '지우', 'peopleNmEn': 'Ji-woo', 'cast': '', 'castEn': ''}], 'showTypes': [{

- 영화 대표코드 : movieCd

- 영화명(국문) : movieNm

- 영화명(영문) : movieNmEn

- 영화명(원문) : movieNmOg

- 개봉연도 : openDt

- 상영시간 : showTm

- 장르 : genreNm

- 감독명 : peopleNm

- 관람등급 : watchGradeNm

- 배우 1, 2, 3 : actors


In [99]:
Movie_info = []

for movie in movie_info:
    Movie_info.append(movie["movieInfoResult"]["movieInfo"])
    
Movie_info

[{'movieCd': '20177538',
  'movieNm': '완벽한 타인',
  'movieNmEn': 'Intimate Strangers',
  'movieNmOg': '',
  'showTm': '115',
  'prdtYear': '2018',
  'openDt': '20181031',
  'prdtStatNm': '개봉',
  'typeNm': '장편',
  'nations': [{'nationNm': '한국'}],
  'genres': [{'genreNm': '드라마'}, {'genreNm': '코미디'}],
  'directors': [{'peopleNm': '이재규', 'peopleNmEn': 'LEE Jae-kyoo'}],
  'actors': [{'peopleNm': '유해진',
    'peopleNmEn': 'YOO Hae-jin',
    'cast': '',
    'castEn': ''},
   {'peopleNm': '조진웅',
    'peopleNmEn': 'CHO Jin-woong',
    'cast': '',
    'castEn': ''},
   {'peopleNm': '이서진',
    'peopleNmEn': 'LEE Seo-jin',
    'cast': '준모',
    'castEn': ''},
   {'peopleNm': '염정아', 'peopleNmEn': 'YUM Jung-ah', 'cast': '', 'castEn': ''},
   {'peopleNm': '김지수', 'peopleNmEn': 'KIM Ji-soo', 'cast': '', 'castEn': ''},
   {'peopleNm': '송하윤', 'peopleNmEn': 'SONG Ha-yoon', 'cast': '', 'castEn': ''},
   {'peopleNm': '윤경호', 'peopleNmEn': '', 'cast': '', 'castEn': ''},
   {'peopleNm': '지우', 'peopleNmEn': 'Ji-wo

In [ ]:
info = []
for info in Movie_info:
    info.append([info["movieCd"], info["movieNm"], info["movieNmEn"], info["movieNmOg"], info["openDt"], info["showTm"], info["genres"][0]["genreNm"]